### PR TITLE
LPD-35412 Prioritize the value localized in the user language

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-35412](https://liferay.atlassian.net/browse/LPD-35412)

[LPD-35412]: https://liferay.atlassian.net/browse/LPD-35412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx [hunk 1]
```diff
@@ -56,8 +56,12 @@ function getLabel<T extends ObjectMap<any>>(
 
 		return value ? String(value) : '';
 	}
+
+	const localizedValue = value as LocalizedValue<string>;
+
 	const label =
-		(value as LocalizedValue<string>)[defaultLanguageId] ??
+		localizedValue[Liferay.ThemeDisplay.getLanguageId()] ??
+		localizedValue[defaultLanguageId] ??
```
Comment (igor-franca) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx:64]: Hey @devpedrotavares, what do think about use `getLocalizableLabel` from `object-js-components-web`?

To do that we need to call `object-js-components-web` in `packege.json`

**Review by igor-franca — CHANGES_REQUESTED** [review]
Nice work @devpedrotavares! Just a comment

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx (hunk 1 above)
Comment (devpedrotavares) [reply modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx]: Done!

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:sf

File: modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx [hunk 2]
```diff
@@ -16,9 +17,9 @@ import {ReactFieldBase as FieldBase} from 'dynamic-data-mapping-form-field-type'
 import {fetch} from 'frontend-js-web';
 import React, {useEffect, useRef, useState} from 'react';
 
-type LocalizedValue<T> = Liferay.Language.LocalizedValue<T>;
+import {Locale} from '../../../../../../../../../dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/types';
```
Comment (igor-franca) [inline modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/js/ObjectRelationship/ObjectRelationship.tsx:20]: Hey @devpedrotavares, could you check if you can use:

`import {Locale} from 'dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/types';
`

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:object

**Comment by igor-franca** [comment]
ci:test:playwright

**Comment by igor-franca** [comment]
ci:test:object

**Review by igor-franca — APPROVED** [review]

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:report:6014768

**Review by gabrielwas — APPROVED** [review]
@devpedrotavares good! 

LGTM ;)

**Comment by devpedrotavares** [comment]
ci:forward

**Pull request closed without merging**